files/__addonLocation__/.eslintrc.js

```diff
@@ -2,7 +2,7 @@
 
 module.exports = {
   root: true,
-  parser: '@babel/eslint-parser',
+  parser: '<%= typescript ? '@typescript-eslint/parser' : '@babel/eslint-parser' %>',
   parserOptions: {
```

Comment:
I think this ember-cli issue is also relevant here: https://github.com/ember-cli/ember-cli/issues/10046

Since both parsers have different options I think the `parserOptions` should also be applied conditionally?

I was looking into fixing that in ember-cli but I'm not sure which options are needed exactly. I found this: https://typescript-eslint.io/docs/linting/typed-linting which sounds nice?

Reply:
Agreed here. Eslint configs need to make heavy use of overrides, because we have a mix of node (cjs), node (esm), and browser (esm, js, ts).

Overrides are way easier to extend, change, target specific files, folders, etc.

It's also how my eslint-configs package can simultaneously support app, addon, v2 addon, js, ts, all from the same config

Reply:
Will wait for https://github.com/ember-cli/ember-cli/pull/10054 to get merged before applying the same fix here.

Reply:
I just applied the changes as in https://github.com/ember-cli/ember-cli/pull/10054. Think that should be good.